### PR TITLE
fix issues flagged by SonarQube

### LINKED ISSUE #8390: sonarqube scan findings
SonarQube flagged the following named parameters as nonexistent; this is holding up the regression tests on GitHub Actions, due to the requirement that all on-prem code be scanned by SonarQube:
https://github.com/spacetelescope/jwst/blob/ec0f0edbf1bd2c31e8522956c1a17aaa0df604be/jwst/ami/lg_model.py#L377
https://github.com/spacetelescope/jwst/blob/ec0f0edbf1bd2c31e8522956c1a17aaa0df604be/jwst/ami/lg_model.py#L481
https://github.com/spacetelescope/jwst/blob/ec0f0edbf1bd2c31e8522956c1a17aaa0df604be/jwst/ami/lg_model.py#L483
https://github.com/spacetelescope/jwst/blob/ec0f0edbf1bd2c31e8522956c1a17aaa0df604be/jwst/ami/lg_model.py#L504-L506
https://github.com/spacetelescope/jwst/blob/ec0f0edbf1bd2c31e8522956c1a17aaa0df604be/jwst/ami/lg_model.py#L517-L518

### PR DESCRIPTION
**Pull request by zacharyburnett**
closes #8390 

This PR addresses issues flagged by SonarQube (https://plsonarqube.stsci.edu/project/issues?resolved=false&inNewCodePeriod=true&types=BUG&id=jwst&open=AY54FDbH6KYUncGRZ448)

**Checklist for maintainers**
- [x] ~added entry in `CHANGES.rst` within the relevant release section~
- [x] updated or added relevant tests
- [x] ~updated relevant documentation~
- [ ] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [x] ~Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)~

### CODE REVIEW

**Comment by zacharyburnett** [comment]
regtest running at https://plwishmaster.stsci.edu:8081/blue/organizations/jenkins/RT%2FJWST-Developers-Pull-Requests/detail/JWST-Developers-Pull-Requests/1354/pipeline

**Comment by zacharyburnett** [comment]
SonarScan also found some issues here: #8390

File: jwst/ami/oifits.py [hunk 1]
```diff
@@ -579,10 +579,6 @@ def calibrate(self):
         self.calib_oimodel.vis["VISAMP"] = va_out
         self.calib_oimodel.vis["VISAMPERR"] = vaerr_out
 
-        self.calib_oimodel.array["PISTON_T"] = pistons_t
-        self.calib_oimodel.array["PISTON_C"] = pistons_c
-        self.calib_oimodel.array["PIST_ERR"] = pisterr_out
-
         # add calibrated header keywords
```
Comment (tapastro) [inline jwst/ami/oifits.py:582]: Kneejerk response was "this change doesn't look reasonable", but I see these lines are duplicates of L570:572. I'll wait for regtests, but changes look good.

**Comment by zacharyburnett** [comment]
regtests at 90% with no failures

**Comment by braingram** [comment]
Thanks for opening the PR. As all the changes are in the new niriss AMI code and haven't so far triggered errors (which they should given the types of errors) I expect they are uncovered. Indeed I see no call to `fit_image`.

@rcooper295 is `NrmModel.fit_image` used?
https://github.com/spacetelescope/jwst/blob/ec0f0edbf1bd2c31e8522956c1a17aaa0df604be/jwst/ami/lg_model.py#L292

If not I'd say we remove the method.

Also the regtests will need to be run against CRDS-TEST to test the AMI code until ops is updated. It looks like the current run is using ops so the AMI tests are expected to fail.

**Review by braingram — APPROVED** [review]
I ran the AMI regtests and unit tests with this branch locally (with a cache of CRDS-TEST files). The ami regtests pass with these changes.

Perhaps a follow-up issue should be opened to track the wider code cleanup with input from @rcooper295?

**Pull request merged**